### PR TITLE
tests: drivers: Added overlays to tests for 063_ble

### PR DESCRIPTION
**Pull request by pavloinfineon**
Update overlay for adc
Added overlay for gpio_basic_api
Added overlay for i2c_api
Updated board.cmake

### CODE REVIEW

**Review by ifyall — DISMISSED** [review]

File: boards/arm/cy8cproto_063_ble/board.cmake [hunk 1]
```diff
@@ -5,3 +5,6 @@
 #***************************************************************************
 
 include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
+
+board_runner_args(pyocd "--target=cy8c6xx7_nosmif")
+include(${ZEPHYR_BASE}/boards/common/pyocd.board.cmake)
```
Comment (MaureenHelm) [inline boards/arm/cy8cproto_063_ble/board.cmake:10]: These changes are unrelated to overlays and need to go in a separate commit.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
```
tests: drivers: Added overlays to tests for 063_ble

Update overlays for adc
Added overlay for gpio_basic_api
Added overlay for i2c_api
Updated board.cmake

Signed-off-by: Pavlo Havrylyuk <pavlo.havrylyuk@infineon.com>
```

Please explain in the commit message **why** you're making these changes

File: boards/arm/cy8cproto_063_ble/board.cmake (hunk 1 above)
Comment (pavloinfineon) [reply boards/arm/cy8cproto_063_ble/board.cmake:10]: Split into 2 commits

**Review by MaureenHelm — APPROVED** [review]

**Review by ifyall — APPROVED** [review]
